object-store-client-common/src/main/scala/uk/gov/hmrc/objectstore/client/ObjectStoreClient.scala

```diff
@@ -37,43 +37,44 @@ class ObjectStoreClient[F[_], REQ_BODY, RES, RES_BODY](
 
   /** Storing an object on an existing path will overwrite the previously stored object on that path. */
   def putObject[CONTENT](
-    path       : Path.File,
-    content    : CONTENT,
+    path: Path.File,
+    content: CONTENT,
     contentType: Option[String] = None,
-    owner      : String         = config.owner
-  )(implicit
-    w: ObjectStoreContentWrite[F, CONTENT, REQ_BODY]
-  ): F[Unit] =
-    F.flatMap(w.writeContent(content, contentType))(c =>
-      F.flatMap(client.put(s"$url/object/$owner/${path.asUri}", c, List(authorizationHeader)))(
-        read.consume
-      )
-    )
+    owner: String               = config.owner
+  )(
+    implicit
+    w: ObjectStoreContentWrite[F, CONTENT, REQ_BODY]): F[Unit] =
+    F.flatMap(w.writeContent(content, contentType))(
+      c =>
+        F.flatMap(client.put(s"$url/object/$owner/${path.asUri}", c, List(authorizationHeader)))(
+          read.consume
+      ))
 
   def getObject[CONTENT](
-    path : Path.File,
-    owner: String    = config.owner
-  )(implicit
-    cr: ObjectStoreContentRead[F, RES_BODY, CONTENT]
-  ): F[Option[Object[CONTENT]]] = {
+    path: Path.File,
+    owner: String = config.owner
+  )(
+    implicit
+    cr: ObjectStoreContentRead[F, RES_BODY, CONTENT]): F[Option[Object[CONTENT]]] = {
     val location = s"$url/object/$owner/${path.asUri}"
     F.flatMap(client.get(location, List(authorizationHeader)))(res =>
-      F.flatMap(read.toObject(location, res)){
+      F.flatMap(read.toObject(location, res)) {
         case Some(obj) => F.map(cr.readContent(obj.content))(c => Some(obj.copy(content = c)))
         case None      => F.pure(None)
-      }
-    )
+    })
   }
 
   def deleteObject(
-    path : Path.File,
-    owner: String    = config.owner
+    path: Path.File,
+    owner: String = config.owner
   ): F[Unit] =
     F.flatMap(client.delete(s"$url/object/$owner/${path.asUri}", List(authorizationHeader)))(read.consume)
 
   def listObjects(
-    path : Path.Directory,
-    owner: String    = config.owner
-  ): F[ObjectListing] =
-    F.flatMap(client.get(s"$url/list/$owner/${path.asUri}", List(authorizationHeader)))(read.toObjectListing)
+    path: Path.Directory,
+    owner: String = config.owner
+  ): F[ObjectListing] = {
+    val effectivePath = if (path.asUri == "/") "" else s"${path.asUri}"
```

Comment:
This is the core change.

Reply:
`else path.asUri`
Can we not do `.stripPrefix("/")` inside the `asUri` function? This would allow us to just trust `path.asUri`to return a consistent Url. It seems that if it is returning "/" or "mydir/" - then it is sometimes returning an absolute path, and sometimes a relative path.

Reply:
Yea. I did that change before which you approved but then I reverted as it actually didn't fix. I'll bring back that change again.